<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Data Analysis Of Haemapoetic Stem Cell Linages</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="README_files/libs/clipboard/clipboard.min.js"></script>
<script src="README_files/libs/quarto-html/quarto.js"></script>
<script src="README_files/libs/quarto-html/popper.min.js"></script>
<script src="README_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="README_files/libs/quarto-html/anchor.min.js"></script>
<link href="README_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="README_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="README_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="README_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="README_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Data Analysis Of Haemapoetic Stem Cell Linages</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/Haemopetic%20stem%20cell%20heirarchy.png" width="607" height="243" class="figure-img"></p>
</figure>
</div>
<p>Documentation for the project “Data Analysis of Hematopoietic Stem Cell Lineages,” detailing the exploration of the differentiation hierarchy of hematopoietic stem cells (HSCs) through single-cell RNA-sequencing data and confriming assumtions using quantatative PCR.</p>
<section id="project-description" class="level2">
<h2 class="anchored" data-anchor-id="project-description">Project Description</h2>
<p>Hematopoietic stem cells (HSCs) are at the apex of the differentiation hierarchy and defined by their ability to differentiate and produces the full spectrum of mature blood cells and self-renew to sustain haematopoiesis. Long-term hematopoietic stem cells (LT.HSCs) sit within this HSC pool and are of much interest due to their capacity to support long-term hematopoietic reconstitution <span class="citation" data-cites="gur-cohen2016 notta2011">(<a href="#ref-gur-cohen2016" role="doc-biblioref">Gur-Cohen et al. 2016</a>; <a href="#ref-notta2011" role="doc-biblioref">Notta et al. 2011</a>)</span>. Utilising surface markers for these cells has allowed for HSC purities of more than 50%. However, none have allowed for populations of a single homogeneous cell type <span class="citation" data-cites="adam2005 kent2009 zhang2005 nestorowa2016">(<a href="#ref-adam2005" role="doc-biblioref">Adam et al. 2005</a>; <a href="#ref-kent2009" role="doc-biblioref">Kent et al. 2009</a>; <a href="#ref-zhang2005" role="doc-biblioref">Zhang and Lodish 2005</a>; <a href="#ref-nestorowa2016" role="doc-biblioref">Nestorowa et al. 2016</a>)</span>.</p>
<p>Recently there has been the development of single-cell profiling techniques which are able to profile transitional cells and resolve the heterogeneity within these populations’ at large numbers <span class="citation" data-cites="bendall2014 jaitin2014 mahata2014">(<a href="#ref-bendall2014" role="doc-biblioref">Bendall et al. 2014</a>; <a href="#ref-jaitin2014" role="doc-biblioref">Jaitin et al. 2014</a>; <a href="#ref-mahata2014" role="doc-biblioref">Mahata et al. 2014</a>)</span>. This should allow the identification of novel HSC markers <span class="citation" data-cites="nestorowa2016">(<a href="#ref-nestorowa2016" role="doc-biblioref">Nestorowa et al. 2016</a>)</span>.</p>
<p>We utilised single cell RNA-sequencing data collected by <span class="citation" data-cites="nestorowa2016">(<a href="#ref-nestorowa2016" role="doc-biblioref">Nestorowa et al. 2016</a>)</span> which analysed over 1600 single HSPC transcriptomes which were profiled into LT-HSCs, HSPCs and Progenitor cells using FACS index sorting and using wide gating to allow for analysis of transitional cells. This allowed us to find a new marker of LT.HSCs.</p>
</section>
<section id="dates" class="level2">
<h2 class="anchored" data-anchor-id="dates">Dates</h2>
<ul>
<li><strong>Start Date</strong>: 05/04/24</li>
<li><strong>Last Updated Date</strong>: 07/04/24</li>
</ul>
</section>
<section id="contact-information" class="level2">
<h2 class="anchored" data-anchor-id="contact-information">Contact Information</h2>
<ul>
<li><strong>Author</strong>: Joseph Womersley</li>
<li><strong>Email</strong>: Joe.Womersley02@gmail.com</li>
</ul>
</section>
<section id="folder-structure" class="level2">
<h2 class="anchored" data-anchor-id="folder-structure">Folder Structure</h2>
</section>
<section id="software-requirements" class="level2">
<h2 class="anchored" data-anchor-id="software-requirements">Software Requirements</h2>
<section id="r-environment" class="level3">
<h3 class="anchored" data-anchor-id="r-environment">R Environment</h3>
<ul>
<li><strong>R Version</strong>: 4.1.0</li>
<li><strong>Packages</strong>:
<ul>
<li>tidyverse (1.3.0): For data manipulation and visualization. (Wickham et al., 2019)</li>
<li>readxl (1.3.1): For reading Excel files. (Wickham and Bryan, 2019)</li>
<li>pheatmap (1.0.12): For generating heatmaps. (Kolde, 2019)</li>
<li>Plotly (4.9.2.1): For interactive visualizations. (Sievert, 2020)</li>
<li>sessioninfo (1.1.1): For reporting R session information.</li>
</ul></li>
</ul>
<p>References cited within the project report itself are crucial for the report’s academic integrity and understanding but are not necessarily included in the README unless they directly inform the README’s content.</p>

</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-adam2005" class="csl-entry" role="listitem">
Adam, Frédérick, Shilun Zheng, Nilesh Joshi, David S. Kelton, Amin Sandhu, Youko Suehiro, Samira B. Jeimy, et al. 2005. <span>“Analyses of cellular multimerin 1 receptors: in vitro evidence of binding mediated by alphaIIbbeta3 and alphavbeta3.”</span> <em>Thrombosis and Haemostasis</em> 94 (5): 1004–11. <a href="https://doi.org/10.1160/TH05-02-0140">https://doi.org/10.1160/TH05-02-0140</a>.
</div>
<div id="ref-bendall2014" class="csl-entry" role="listitem">
Bendall, Sean C., Kara L. Davis, El-Ad David Amir, Michelle D. Tadmor, Erin F. Simonds, Tiffany J. Chen, Daniel K. Shenfeld, Garry P. Nolan, and Dana Pe’er. 2014. <span>“Single-cell trajectory detection uncovers progression and regulatory coordination in human B cell development.”</span> <em>Cell</em> 157 (3): 714–25. <a href="https://doi.org/10.1016/j.cell.2014.04.005">https://doi.org/10.1016/j.cell.2014.04.005</a>.
</div>
<div id="ref-gur-cohen2016" class="csl-entry" role="listitem">
Gur-Cohen, Shiri, Orit Kollet, Claudine Graf, Charles T. Esmon, Wolfram Ruf, and Tsvee Lapidot. 2016. <span>“Regulation of Long-Term Repopulating Hematopoietic Stem Cells by EPCR/PAR1 Signaling.”</span> <em>Annals of the New York Academy of Sciences</em> 1370 (1): 65–81. <a href="https://doi.org/10.1111/nyas.13013">https://doi.org/10.1111/nyas.13013</a>.
</div>
<div id="ref-jaitin2014" class="csl-entry" role="listitem">
Jaitin, Diego Adhemar, Ephraim Kenigsberg, Hadas Keren-Shaul, Naama Elefant, Franziska Paul, Irina Zaretsky, Alexander Mildner, et al. 2014. <span>“Massively parallel single-cell RNA-seq for marker-free decomposition of tissues into cell types.”</span> <em>Science (New York, N.Y.)</em> 343 (6172): 776–79. <a href="https://doi.org/10.1126/science.1247651">https://doi.org/10.1126/science.1247651</a>.
</div>
<div id="ref-kent2009" class="csl-entry" role="listitem">
Kent, David G., Michael R. Copley, Claudia Benz, Stefan Wöhrer, Brad J. Dykstra, Elaine Ma, John Cheyne, et al. 2009. <span>“Prospective Isolation and Molecular Characterization of Hematopoietic Stem Cells with Durable Self-Renewal Potential.”</span> <em>Blood</em> 113 (25): 6342–50. <a href="https://doi.org/10.1182/blood-2008-12-192054">https://doi.org/10.1182/blood-2008-12-192054</a>.
</div>
<div id="ref-mahata2014" class="csl-entry" role="listitem">
Mahata, Bidesh, Xiuwei Zhang, Aleksandra A. Kolodziejczyk, Valentina Proserpio, Liora Haim-Vilmovsky, Angela E. Taylor, Daniel Hebenstreit, et al. 2014. <span>“Single-cell RNA sequencing reveals T helper cells synthesizing steroids de novo to contribute to immune homeostasis.”</span> <em>Cell Reports</em> 7 (4): 1130–42. <a href="https://doi.org/10.1016/j.celrep.2014.04.011">https://doi.org/10.1016/j.celrep.2014.04.011</a>.
</div>
<div id="ref-nestorowa2016" class="csl-entry" role="listitem">
Nestorowa, Sonia, Fiona K. Hamey, Blanca Pijuan Sala, Evangelia Diamanti, Mairi Shepherd, Elisa Laurenti, Nicola K. Wilson, David G. Kent, and Berthold Göttgens. 2016. <span>“A Single-Cell Resolution Map of Mouse Hematopoietic Stem and Progenitor Cell Differentiation.”</span> <em>Blood</em> 128 (8): e20–31. <a href="https://doi.org/10.1182/blood-2016-05-716480">https://doi.org/10.1182/blood-2016-05-716480</a>.
</div>
<div id="ref-notta2011" class="csl-entry" role="listitem">
Notta, Faiyaz, Sergei Doulatov, Elisa Laurenti, Armando Poeppl, Igor Jurisica, and John E. Dick. 2011. <span>“Isolation of single human hematopoietic stem cells capable of long-term multilineage engraftment.”</span> <em>Science (New York, N.Y.)</em> 333 (6039): 218–21. <a href="https://doi.org/10.1126/science.1201219">https://doi.org/10.1126/science.1201219</a>.
</div>
<div id="ref-zhang2005" class="csl-entry" role="listitem">
Zhang, Cheng Cheng, and Harvey F. Lodish. 2005. <span>“Murine hematopoietic stem cells change their surface phenotype during ex vivo expansion.”</span> <em>Blood</em> 105 (11): 4314–20. <a href="https://doi.org/10.1182/blood-2004-11-4418">https://doi.org/10.1182/blood-2004-11-4418</a>.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>