<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Data Analysis Of Haemapoetic Stem Cell Linages</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="README_files/libs/clipboard/clipboard.min.js"></script>
<script src="README_files/libs/quarto-html/quarto.js"></script>
<script src="README_files/libs/quarto-html/popper.min.js"></script>
<script src="README_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="README_files/libs/quarto-html/anchor.min.js"></script>
<link href="README_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="README_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="README_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="README_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="README_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Data Analysis Of Haemapoetic Stem Cell Linages</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<div style="text-align: left;">
  <img src="images/Haemopetic%20stem%20cell%20heirarchy..png" width="795" height="270" alt="Haemopoietic Stem Cell Hierarchy">
  <p style="color: gray; font-size: 0.6em;">Laurenti, E. and Göttgens, B. (2018) ‘From haematopoietic stem cells to complex differentiation landscapes’, Nature, 553(7689), pp. 418–426. Available at: https://doi.org/10.1038/nature25022.</p>
</div>
<p>Documentation for the project “Data Analysis of Hematopoietic Stem Cell Lineages,” detailing the exploration of the differentiation hierarchy of hematopoietic stem cells (HSCs) through single-cell RNA-sequencing data and confriming assumtions using quantatative PCR.</p>
<section id="project-description" class="level2">
<h2 class="anchored" data-anchor-id="project-description">Project Description</h2>
<p>Hematopoietic stem cells (HSCs) are at the apex of the differentiation hierarchy and defined by their ability to differentiate and produces the full spectrum of mature blood cells and self-renew to sustain haematopoiesis. Long-term hematopoietic stem cells (LT.HSCs) sit within this HSC pool and are of much interest due to their capacity to support long-term hematopoietic reconstitution <span class="citation" data-cites="gur-cohen2016 notta2011">(<a href="#ref-gur-cohen2016" role="doc-biblioref">Gur-Cohen et al. 2016</a>; <a href="#ref-notta2011" role="doc-biblioref">Notta et al. 2011</a>)</span>. Utilising surface markers for these cells has allowed for HSC purities of more than 50%. However, none have allowed for populations of a single homogeneous cell type <span class="citation" data-cites="adam2005 kent2009 zhang2005 nestorowa2016">(<a href="#ref-adam2005" role="doc-biblioref">Adam et al. 2005</a>; <a href="#ref-kent2009" role="doc-biblioref">Kent et al. 2009</a>; <a href="#ref-zhang2005" role="doc-biblioref">Zhang and Lodish 2005</a>; <a href="#ref-nestorowa2016" role="doc-biblioref">Nestorowa et al. 2016</a>)</span>.</p>
<p>Recently there has been the development of single-cell profiling techniques which are able to profile transitional cells and resolve the heterogeneity within these populations’ at large numbers <span class="citation" data-cites="bendall2014 jaitin2014 mahata2014">(<a href="#ref-bendall2014" role="doc-biblioref">Bendall et al. 2014</a>; <a href="#ref-jaitin2014" role="doc-biblioref">Jaitin et al. 2014</a>; <a href="#ref-mahata2014" role="doc-biblioref">Mahata et al. 2014</a>)</span>. This should allow the identification of novel HSC markers <span class="citation" data-cites="nestorowa2016">(<a href="#ref-nestorowa2016" role="doc-biblioref">Nestorowa et al. 2016</a>)</span>.</p>
<p>We utilised single cell RNA-sequencing data collected by <span class="citation" data-cites="nestorowa2016">(<a href="#ref-nestorowa2016" role="doc-biblioref">Nestorowa et al. 2016</a>)</span> which analysed over 1600 single HSPC transcriptomes which were profiled into LT-HSCs, HSPCs and Progenitor cells using FACS index sorting and using wide gating to allow for analysis of transitional cells. This allowed us to find a potential new marker of LT.HSCs.</p>
</section>
<section id="installation" class="level2">
<h2 class="anchored" data-anchor-id="installation">Installation</h2>
<p><code>REPORT::</code> can be installed using <a href="https://devtools.r-lib.org"><code>devtools::</code></a> from GitHub with:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>usethis<span class="sc">::</span><span class="fu">use_course</span>(<span class="at">url =</span> <span class="st">"JoeWomersley/REPORT"</span>, <span class="at">destdir =</span> <span class="st">"."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="regenerating-the-analysis" class="level2">
<h2 class="anchored" data-anchor-id="regenerating-the-analysis">Regenerating The Analysis:</h2>
<p>The analysis can freely reproduced and adapted by simply:</p>
<ol type="1">
<li><p>Download <a href="https://www.r-project.org">R studio and R version</a>: 4.3.1 (2023-06-16)</p></li>
<li><p>follow the <a href="#installation">Installation</a> instructions above</p></li>
<li><p>Install <a href="#packages">Packages</a>, listed below</p></li>
<li><p>Rendering within the Quarto framework, leveraging the integrated <strong>code execution engine</strong>.</p></li>
<li><p>This should allow interaction with the plots using the html output.</p></li>
<li><p>If you desire to change the raw data set, simply alter the data-raw file and change the <code>Read.csv</code> function to the new file path. I would highlight that the Analysis does include some hard coding to alleviate some issues, where hard coding is used there should be comments within the code highlighting this.</p></li>
</ol>
</section>
<section id="dates" class="level2">
<h2 class="anchored" data-anchor-id="dates">Dates</h2>
<ul>
<li><strong>Start Date</strong>: 05/04/24</li>
<li><strong>Last Updated Date</strong>: 07/04/24</li>
</ul>
</section>
<section id="contact-information" class="level2">
<h2 class="anchored" data-anchor-id="contact-information">Contact Information</h2>
<ul>
<li><strong>Author</strong>: Joseph Womersley</li>
<li><strong>Email</strong>: Joe.Womersley02@gmail.com</li>
</ul>
</section>
<section id="technical-description" class="level2">
<h2 class="anchored" data-anchor-id="technical-description">technical description</h2>
<p><strong>R Version</strong>: 4.3.1 (2023-06-16) <span class="citation" data-cites="base">(<a href="#ref-base" role="doc-biblioref">2023</a>)</span></p>
</section>
<section id="contributions" class="level2">
<h2 class="anchored" data-anchor-id="contributions">contributions</h2>
<p>All of the analysis was written with the assistence of Co-pilot <span class="citation" data-cites="githubc2024a">(<a href="#ref-githubc2024a" role="doc-biblioref"><span>“GitHub Copilot · Your AI Pair Programmer”</span> 2024</a>)</span> and chatGPT <span class="citation" data-cites="chatgpt2024">(<a href="#ref-chatgpt2024" role="doc-biblioref"><span>“ChatGPT”</span> 2024</a>)</span></p>
</section>
<section id="packages" class="level2">
<h2 class="anchored" data-anchor-id="packages">Packages</h2>
<ul>
<li><strong>Tidyverse (1.3.0)</strong>: A collection of R packages designed for data science, offering tools for data manipulation, analysis, and visualization. Includes <strong><code>dplyr</code></strong> for data manipulation and <strong><code>ggplot2</code></strong> for visualization. <span class="citation" data-cites="tidyverse">(<a href="#ref-tidyverse" role="doc-biblioref">Wickham et al. 2019</a>)</span></li>
<li><strong>Patchwork (1.1.3)</strong>: Combines multiple ggplot2 plots into a cohesive figure, enhancing data presentation and interpretation. <span class="citation" data-cites="patchwork">(<a href="#ref-patchwork" role="doc-biblioref">Pedersen 2023</a>)</span></li>
<li><strong>HtmlTable (2.4.2)</strong> and <strong>DT (0.31)</strong>: Essential for creating interactive HTML and dynamic tables within Quarto. <span class="citation" data-cites="htmlTable DT">(<a href="#ref-htmlTable" role="doc-biblioref">Gordon, Gragg, and Konings 2023</a>; <a href="#ref-DT" role="doc-biblioref">Xie, Cheng, and Tan 2023</a>)</span></li>
<li><strong>Conflicted (1.2.0)</strong>: To manage namespace conflicts between loaded packages. <span class="citation" data-cites="conflicted">(<a href="#ref-conflicted" role="doc-biblioref">Wickham 2023</a>)</span></li>
<li><strong>Scran (1.28.2)</strong> and <strong>BiomaRt (2.56.1)</strong>: For differential analysis of the single-cell RNA-seq data and gene annotations. <span class="citation" data-cites="scran biomaRt">(<a href="#ref-scran" role="doc-biblioref">Lun, McCarthy, and Marioni 2016</a>; <a href="#ref-biomaRt" role="doc-biblioref">Durinck et al. 2009</a>)</span></li>
<li><strong>Bookdown (0.38)</strong>, <strong>Knitr (1.45)</strong>, and <strong>KableExtra (1.4.0)</strong>: Facilitate the creation of the documents and report, enabling comprehensive documentation of the analysis process. <span class="citation" data-cites="bookdown knitr kableExtra">(<a href="#ref-bookdown" role="doc-biblioref">Xie 2024</a>, <a href="#ref-knitr" role="doc-biblioref">2023</a>; <a href="#ref-kableExtra" role="doc-biblioref">Zhu 2024</a>)</span></li>
<li><strong>ggrepel (0.9.4)</strong>: Improves clarity in ggplot2 visualizations by preventing text label overlaps. <span class="citation" data-cites="ggrepel">(<a href="#ref-ggrepel" role="doc-biblioref">Slowikowski 2023</a>)</span></li>
<li><strong>Pheatmap (1.0.12)</strong>: For creating heatmaps with more control over aesthetics. <span class="citation" data-cites="pheatmap">(<a href="#ref-pheatmap" role="doc-biblioref">Kolde 2019</a>)</span></li>
<li><strong>Plotly (4.10.4.9000)</strong>: Enables interactive plots that can be embedded in web apps or documents. <span class="citation" data-cites="plotly">(<a href="#ref-plotly" role="doc-biblioref">Sievert 2020</a>)</span></li>
<li><strong>Readxl (1.4.3)</strong>: Simplifies the process of importing Excel data into R, facilitating data integration from diverse sources. <span class="citation" data-cites="readxl">(<a href="#ref-readxl" role="doc-biblioref">Wickham and Bryan 2023</a>)</span></li>
</ul>

</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-adam2005" class="csl-entry" role="listitem">
Adam, Frédérick, Shilun Zheng, Nilesh Joshi, David S. Kelton, Amin Sandhu, Youko Suehiro, Samira B. Jeimy, et al. 2005. <span>“Analyses of cellular multimerin 1 receptors: in vitro evidence of binding mediated by alphaIIbbeta3 and alphavbeta3.”</span> <em>Thrombosis and Haemostasis</em> 94 (5): 1004–11. <a href="https://doi.org/10.1160/TH05-02-0140">https://doi.org/10.1160/TH05-02-0140</a>.
</div>
<div id="ref-bendall2014" class="csl-entry" role="listitem">
Bendall, Sean C., Kara L. Davis, El-Ad David Amir, Michelle D. Tadmor, Erin F. Simonds, Tiffany J. Chen, Daniel K. Shenfeld, Garry P. Nolan, and Dana Pe’er. 2014. <span>“Single-cell trajectory detection uncovers progression and regulatory coordination in human B cell development.”</span> <em>Cell</em> 157 (3): 714–25. <a href="https://doi.org/10.1016/j.cell.2014.04.005">https://doi.org/10.1016/j.cell.2014.04.005</a>.
</div>
<div id="ref-chatgpt2024" class="csl-entry" role="listitem">
<span>“ChatGPT.”</span> 2024. <a href="https://chat.openai.com">https://chat.openai.com</a>.
</div>
<div id="ref-biomaRt" class="csl-entry" role="listitem">
Durinck, Steffen, Paul T. Spellman, Ewan Birney, and Wolfgang Huber. 2009. <span>“Mapping Identifiers for the Integration of Genomic Datasets with the r/Bioconductor Package biomaRt”</span> 4.
</div>
<div id="ref-githubc2024a" class="csl-entry" role="listitem">
<span>“GitHub Copilot · Your AI Pair Programmer.”</span> 2024. <a href="https://github.com/features/copilot">https://github.com/features/copilot</a>.
</div>
<div id="ref-htmlTable" class="csl-entry" role="listitem">
Gordon, Max, Stephen Gragg, and Peter Konings. 2023. <span>“htmlTable: Advanced Tables for Markdown/HTML.”</span> <a href="https://CRAN.R-project.org/package=htmlTable">https://CRAN.R-project.org/package=htmlTable</a>.
</div>
<div id="ref-gur-cohen2016" class="csl-entry" role="listitem">
Gur-Cohen, Shiri, Orit Kollet, Claudine Graf, Charles T. Esmon, Wolfram Ruf, and Tsvee Lapidot. 2016. <span>“Regulation of Long-Term Repopulating Hematopoietic Stem Cells by EPCR/PAR1 Signaling.”</span> <em>Annals of the New York Academy of Sciences</em> 1370 (1): 65–81. <a href="https://doi.org/10.1111/nyas.13013">https://doi.org/10.1111/nyas.13013</a>.
</div>
<div id="ref-jaitin2014" class="csl-entry" role="listitem">
Jaitin, Diego Adhemar, Ephraim Kenigsberg, Hadas Keren-Shaul, Naama Elefant, Franziska Paul, Irina Zaretsky, Alexander Mildner, et al. 2014. <span>“Massively parallel single-cell RNA-seq for marker-free decomposition of tissues into cell types.”</span> <em>Science (New York, N.Y.)</em> 343 (6172): 776–79. <a href="https://doi.org/10.1126/science.1247651">https://doi.org/10.1126/science.1247651</a>.
</div>
<div id="ref-kent2009" class="csl-entry" role="listitem">
Kent, David G., Michael R. Copley, Claudia Benz, Stefan Wöhrer, Brad J. Dykstra, Elaine Ma, John Cheyne, et al. 2009. <span>“Prospective Isolation and Molecular Characterization of Hematopoietic Stem Cells with Durable Self-Renewal Potential.”</span> <em>Blood</em> 113 (25): 6342–50. <a href="https://doi.org/10.1182/blood-2008-12-192054">https://doi.org/10.1182/blood-2008-12-192054</a>.
</div>
<div id="ref-pheatmap" class="csl-entry" role="listitem">
Kolde, Raivo. 2019. <span>“Pheatmap: Pretty Heatmaps.”</span> <a href="https://CRAN.R-project.org/package=pheatmap">https://CRAN.R-project.org/package=pheatmap</a>.
</div>
<div id="ref-scran" class="csl-entry" role="listitem">
Lun, Aaron T. L., Davis J. McCarthy, and John C. Marioni. 2016. <span>“A Step-by-Step Workflow for Low-Level Analysis of Single-Cell RNA-Seq Data with Bioconductor”</span> 5: 2122. <a href="https://doi.org/10.12688/f1000research.9501.2">https://doi.org/10.12688/f1000research.9501.2</a>.
</div>
<div id="ref-mahata2014" class="csl-entry" role="listitem">
Mahata, Bidesh, Xiuwei Zhang, Aleksandra A. Kolodziejczyk, Valentina Proserpio, Liora Haim-Vilmovsky, Angela E. Taylor, Daniel Hebenstreit, et al. 2014. <span>“Single-cell RNA sequencing reveals T helper cells synthesizing steroids de novo to contribute to immune homeostasis.”</span> <em>Cell Reports</em> 7 (4): 1130–42. <a href="https://doi.org/10.1016/j.celrep.2014.04.011">https://doi.org/10.1016/j.celrep.2014.04.011</a>.
</div>
<div id="ref-nestorowa2016" class="csl-entry" role="listitem">
Nestorowa, Sonia, Fiona K. Hamey, Blanca Pijuan Sala, Evangelia Diamanti, Mairi Shepherd, Elisa Laurenti, Nicola K. Wilson, David G. Kent, and Berthold Göttgens. 2016. <span>“A Single-Cell Resolution Map of Mouse Hematopoietic Stem and Progenitor Cell Differentiation.”</span> <em>Blood</em> 128 (8): e20–31. <a href="https://doi.org/10.1182/blood-2016-05-716480">https://doi.org/10.1182/blood-2016-05-716480</a>.
</div>
<div id="ref-notta2011" class="csl-entry" role="listitem">
Notta, Faiyaz, Sergei Doulatov, Elisa Laurenti, Armando Poeppl, Igor Jurisica, and John E. Dick. 2011. <span>“Isolation of single human hematopoietic stem cells capable of long-term multilineage engraftment.”</span> <em>Science (New York, N.Y.)</em> 333 (6039): 218–21. <a href="https://doi.org/10.1126/science.1201219">https://doi.org/10.1126/science.1201219</a>.
</div>
<div id="ref-patchwork" class="csl-entry" role="listitem">
Pedersen, Thomas Lin. 2023. <span>“Patchwork: The Composer of Plots.”</span> <a href="https://CRAN.R-project.org/package=patchwork">https://CRAN.R-project.org/package=patchwork</a>.
</div>
<div id="ref-base" class="csl-entry" role="listitem">
R Core Team. 2023. <span>“R: A Language and Environment for Statistical Computing.”</span> <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-plotly" class="csl-entry" role="listitem">
Sievert, Carson. 2020. <span>“Interactive Web-Based Data Visualization with r, Plotly, and Shiny.”</span> <a href="https://plotly-r.com">https://plotly-r.com</a>.
</div>
<div id="ref-ggrepel" class="csl-entry" role="listitem">
Slowikowski, Kamil. 2023. <span>“Ggrepel: Automatically Position Non-Overlapping Text Labels with ’Ggplot2’.”</span> <a href="https://CRAN.R-project.org/package=ggrepel">https://CRAN.R-project.org/package=ggrepel</a>.
</div>
<div id="ref-conflicted" class="csl-entry" role="listitem">
Wickham, Hadley. 2023. <span>“Conflicted: An Alternative Conflict Resolution Strategy.”</span> <a href="https://CRAN.R-project.org/package=conflicted">https://CRAN.R-project.org/package=conflicted</a>.
</div>
<div id="ref-tidyverse" class="csl-entry" role="listitem">
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. <span>“Welcome to the <span></span>Tidyverse<span></span>”</span> 4: 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.
</div>
<div id="ref-readxl" class="csl-entry" role="listitem">
Wickham, Hadley, and Jennifer Bryan. 2023. <span>“Readxl: Read Excel Files.”</span> <a href="https://CRAN.R-project.org/package=readxl">https://CRAN.R-project.org/package=readxl</a>.
</div>
<div id="ref-knitr" class="csl-entry" role="listitem">
Xie, Yihui. 2023. <span>“Knitr: A General-Purpose Package for Dynamic Report Generation in r.”</span> <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
<div id="ref-bookdown" class="csl-entry" role="listitem">
———. 2024. <span>“Bookdown: Authoring Books and Technical Documents with r Markdown.”</span> <a href="https://github.com/rstudio/bookdown">https://github.com/rstudio/bookdown</a>.
</div>
<div id="ref-DT" class="csl-entry" role="listitem">
Xie, Yihui, Joe Cheng, and Xianying Tan. 2023. <span>“DT: A Wrapper of the JavaScript Library ’DataTables’.”</span> <a href="https://CRAN.R-project.org/package=DT">https://CRAN.R-project.org/package=DT</a>.
</div>
<div id="ref-zhang2005" class="csl-entry" role="listitem">
Zhang, Cheng Cheng, and Harvey F. Lodish. 2005. <span>“Murine hematopoietic stem cells change their surface phenotype during ex vivo expansion.”</span> <em>Blood</em> 105 (11): 4314–20. <a href="https://doi.org/10.1182/blood-2004-11-4418">https://doi.org/10.1182/blood-2004-11-4418</a>.
</div>
<div id="ref-kableExtra" class="csl-entry" role="listitem">
Zhu, Hao. 2024. <span>“kableExtra: Construct Complex Table with ’Kable’ and Pipe Syntax.”</span> <a href="https://CRAN.R-project.org/package=kableExtra">https://CRAN.R-project.org/package=kableExtra</a>.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>